ext/drx/drx_time_scale.c

```diff
@@ -216,6 +221,40 @@ inflate_timespec(void *drcontext, struct timespec *spec, int scale,
            dr_get_thread_id(drcontext), scale, spec->tv_sec, spec->tv_nsec);
 }
 
+static bool
+inflate_absolute_timespec(void *drcontext, struct timespec *spec, int scale,
+                          drx_time_scale_type_t type, bool realtime)
+{
+    /* First, convert to relative. */
+    struct timespec cur_time;
+    int res =
+        dr_invoke_syscall_as_app(drcontext, SYS_clock_gettime, 2,
+                                 realtime ? CLOCK_REALTIME : CLOCK_MONOTONIC, &cur_time);
+    if (res != 0) {
+        NOTIFY(0, "Failed to call clock_gettime\n");
+        increment_attempt_and_failure(type);
+        return false;
+    }
+    int64_t cur_nanos = (int64_t)cur_time.tv_sec * MAX_TV_NSEC + cur_time.tv_nsec;
+    int64_t target_nanos = (int64_t)spec->tv_sec * MAX_TV_NSEC + spec->tv_nsec;
+    int64_t diff_nanos = target_nanos < cur_nanos ? 0 : target_nanos - cur_nanos;
+    if (diff_nanos == 0) {
+        diff_nanos = ZERO_PRE_INFLATE_NSEC;
```

Comment:
Update comment by ZERO_PRE_INFLATE_NSEC noting that it applies to all "blocking" syscalls that may accept a "no-op"-like duration value.